tests/integration_tests/upgrade_gc_safepoint/run.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+set -e
```

Comment:
```suggestion
set -eu
```

I hope we can switch to `set -eu`, see https://github.com/pingcap/tiflow/pull/4160 .

also the problem of wrong position of `pd_cluster_id` in `gc_safepoint` can be found by it https://github.com/pingcap/tiflow/pull/4160#issuecomment-1002933474

Reply:
Please go on with #4160, this PR will not be merged until #4115 is fixed.